src/core/qgsvectorfilewriter.cpp

```diff
@@ -3163,46 +3163,44 @@ double QgsVectorFileWriter::mapUnitScaleFactor( double scale, QgsUnitTypes::Rend
 
 void QgsVectorFileWriter::startRender( QgsVectorLayer *vl )
 {
-  QgsFeatureRenderer *renderer = symbologyRenderer( vl );
-  if ( !renderer )
+  mRenderer = createSymbologyRenderer( vl );
+  if ( !mRenderer )
   {
     return;
   }
 
-  renderer->startRender( mRenderContext, vl->fields() );
+  mRenderer->startRender( mRenderContext, vl->fields() );
 }
 
-void QgsVectorFileWriter::stopRender( QgsVectorLayer *vl )
+void QgsVectorFileWriter::stopRender()
```

Comment:
What do you think about a `QThread` member in debug mode and add some ASSERT's to check for unsafe usage?

Reply:
Good idea!